macros/edr/tests/test_exposure_schema_validity.sql

```diff
@@ -0,0 +1,58 @@
+    {%- set full_table_name = elementary.relation_to_full_name(model_relation) -%}
+    {{- elementary.test_log('start', full_table_name, 'exposure validation') -}}
+
+    {%- set matching_exposures = [] -%}
+
+    {%- for exposure in (exposures|default(graph.exposures)).values() -%}
+        {%- if node | default(context['model']['attached_node']) in exposure.depends_on.nodes and exposure['meta'] | default(none) is not none -%}
+            {%- do matching_exposures.append(exposure) -%}
+        {%- endif -%}
+    {%- endfor -%}
+    {%- set matching_exposures_len = matching_exposures | length -%}
+    {%- if matching_exposures_len > 0 -%}
+        {%- set columns_to_use = columns | default(adapter.get_columns_in_relation(model)) -%}
+        {%- set columns_dict = {} -%}
+        {%- for column in columns_to_use -%}
+            {%- do columns_dict.update({ column['name'] | upper : elementary.normalize_data_type(column['dtype']) }) -%}
+        {%- endfor -%}
+        {%- set invalid_exposures = [] -%}
+        {%- for exposure in matching_exposures -%}
+            {# Depend on meta since column level info is not available on exposures #}
+            {%- set meta = exposure['meta'] | default(none) -%}
+            {%- if meta != none and meta['columns'] | default(none) is iterable -%}
+                {%- for column in meta['columns'] -%}
+                    {%- if column['name'] | upper not in columns_dict.keys() -%}
+                        {%- do invalid_exposures.append({
+                                'exposure': exposure['name'],
+                                'url': exposure['url'],
+                                'error': column['name'] ~ ' column missing in the model'
+                                })
+                        -%}
+                    {%- elif column['data_type'] | default('') != '' and elementary.normalize_data_type(column['data_type']) != columns_dict[column['name'] | upper] -%}
+                        {%- do invalid_exposures.append({
+                                'exposure': exposure['name'],
+                                'url': exposure['url'],
+                                'error': 'different data type for the column ' ~ column['name'] ~ ' ' ~ column['data_type'] ~ ' vs ' ~ columns_dict[column['name'] | upper]
+                                })
+                        -%}
+                    {%- endif -%}
+                {%- endfor -%}
+            {%- endif -%}
+        {%- endfor -%}
+        {%- for invalid_exposure in invalid_exposures %}
+            {{ 'UNION ALL ' if not loop.first }}SELECT '{{ invalid_exposure['exposure'] }}' as exposure, '{{ invalid_exposure['url'] }}' as url, '{{ invalid_exposure['error'] }}' as error
+        {%- endfor -%}
+        {{ elementary.test_log('end', full_table_name, 'exposure validation') }}
+        {% if invalid_exposures | length == 0 %}
+            {{ elementary.no_results_query() }}
+        {% endif %}
```

Comment:
I think it may be more readable to write:
```
{% if invalid_exposures | length > 0 %}
    {%- for invalid_exposure in invalid_exposures %}
        {{ 'UNION ALL ' if not loop.first }}SELECT '{{ invalid_exposure['exposure'] }}' as exposure, '{{invalid_exposure['url'] }}' as url, '{{ invalid_exposure['error'] }}' as error
    {%- endfor -%}
{% else %}
    {{ elementary.no_results_query() }}
{% endif %}
```